app/controlplane/internal/service/attestation.go

```diff
@@ -319,18 +324,23 @@ func extractEnvVariables(in map[string]string) []*cpAPI.AttestationItem_EnvVaria
 	return res
 }
 
-func extractMaterials(in []*chainloop.NormalizedMaterial) []*cpAPI.AttestationItem_Material {
+func extractMaterials(in []*chainloop.NormalizedMaterial) ([]*cpAPI.AttestationItem_Material, error) {
 	res := make([]*cpAPI.AttestationItem_Material, 0, len(in))
 	for _, m := range in {
 		// Initialize simply with the value
 		displayValue := m.Value
 		// Override if there is a hash attached
 		if m.Hash != nil {
-			displayValue = fmt.Sprintf("%s@%s", m.Value, m.Hash)
+			name := m.Value
+			if m.EmbeddedInline || m.UploadedToCAS {
```

Comment:
I think this deserve a separate helper function like is_downloadable.

Reply:
yeah, the problem is that embedded doesn't mean that can be downloaded.

In fact artifact download doesn't work with embedded (inline) artifacts. We'd need to have an artifact-attestation mapping to make it work.

Reply:
I mean, we can call it differently but we already use this "if m.EmbeddedInline || m.UploadedToCAS" in two places.